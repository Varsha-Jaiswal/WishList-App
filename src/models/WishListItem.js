import { types } from 'mobx-state-tree';

export const WishListItem = types
  .model({
    name: types.optional(types.string, ''),
    price: types.optional(types.number, 0),
    image: ''
  })
  .actions(self => ({
    changeName(newName) {
      self.name = newName
    },
    changePrice(newPrice) {
      self.price = newPrice
    }
  }));