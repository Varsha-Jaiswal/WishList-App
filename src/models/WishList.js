
import { types, destroy } from 'mobx-state-tree';
import { WishListItem } from './WishListItem';

export const WishList = types
  .model({
    items: types.optional(types.array(WishListItem), [])
  })
  .actions(self => ({
    add(item) {
      self.items.push(item)
    },
    remove(item) {
      // self.items.splice(self.items.indexOf(item), 1)
      destroy(item) // it just removes an item from the container it lives in.
    }
  }))
  .views(self => ({
    get totalPrice() {
      return self.items.reduce((sum,entry) => sum+entry.price, 0)
    }
  }))