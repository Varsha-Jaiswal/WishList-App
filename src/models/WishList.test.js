import { WishListItem } from "./WishListItem"
import { WishList } from "./WishList";
import { getSnapshot, onSnapshot } from "mobx-state-tree";

it("can create a instance", () => {
  const item = WishListItem.create({
    name: "VarsHa",
    price: 20
  })
  expect(item.price).toBe(20)
  item.changeName("AB")
  expect(item.name).toBe("AB")
})

it("can create a wislist", () => {
  const list = WishList.create({
    items: [
      {
        name: "VarsHa",
        price: 23
      }
    ]
  })
  expect(list.items.length).toBe(1)
  expect(list.items[0].price).toBe(23)
})

it("can add new data" , () => {
  const newData = WishList.create()
  const state = []
  onSnapshot(newData, snapshot => {
    state.push(snapshot)
  })
  newData.add({
      name: "New item",
      price: 40
    })
  expect(newData.items.length).toBe(1)
  newData.items[0].changeName("AB")
  expect(newData.items[0].name).toBe("AB")
  expect(getSnapshot(newData)).toMatchSnapshot()
})