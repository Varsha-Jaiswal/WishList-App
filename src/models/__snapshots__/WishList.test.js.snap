// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new data 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "AB",
      "price": 40,
    },
  ],
}
`;
